presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestMinMaxByNAggregation.java

```diff
@@ -0,0 +1,304 @@
+import com.facebook.presto.operator.aggregation.state.MaxOrMinByState;
+import com.facebook.presto.operator.aggregation.state.MaxOrMinByStateFactory;
+import com.facebook.presto.operator.aggregation.state.MaxOrMinByStateSerializer;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.AbstractFixedWidthType;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.RowType;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slices;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.block.BlockAssertions.createArrayBigintBlock;
+import static com.facebook.presto.block.BlockAssertions.createDoublesBlock;
+import static com.facebook.presto.block.BlockAssertions.createLongRepeatBlock;
+import static com.facebook.presto.block.BlockAssertions.createLongsBlock;
+import static com.facebook.presto.block.BlockAssertions.createRLEBlock;
+import static com.facebook.presto.block.BlockAssertions.createStringsBlock;
+import static com.facebook.presto.operator.aggregation.AggregationTestUtils.assertAggregation;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableSet;
+import static io.airlift.slice.SizeOf.SIZE_OF_DOUBLE;
+import static java.util.Arrays.asList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestMinMaxByNAggregation
+{
+    private static final MetadataManager METADATA = MetadataManager.createTestMetadataManager();
+
+    @Test
+    public void testMaxDoubleDouble()
+            throws Exception
+    {
+        InternalAggregationFunction function = METADATA.getExactFunction(new Signature("max_by", "array<double>", StandardTypes.DOUBLE, StandardTypes.DOUBLE, StandardTypes.BIGINT)).getAggregationFunction();
+        assertAggregation(
+                function,
+                1.0,
+                Arrays.asList((Double) null),
+                createDoublesBlock(1.0, null),
+                createDoublesBlock(3.0, 5.0),
+                createRLEBlock(1L, 2));
+
+        assertAggregation(
+                function,
+                1.0,
+                null,
+                createDoublesBlock(null, null),
+                createDoublesBlock(null, null),
+                createRLEBlock(1L, 2));
+
```

Comment:
Add a test case where n=2. There are 4 input key-value pairs. But 3 of them have null key.
Also add a test case where n=2. But there is only 1 input key-value pair.